core/src/main/java/hudson/slaves/Channels.java

```diff
@@ -223,7 +223,7 @@ public static Channel newJVM(String displayName, TaskListener listener, JVMBuild
         vmb.mainClass(Launcher.class);
 
         if (classpath != null)
-            vmb.args().add("-cp").add(classpath);
+            java.util.Arrays.stream(classpath.toString().split(java.io.File.pathSeparator)).forEach(arg -> vmb.classpath().add(arg));
```

Comment:
Note use of `pathSeparator` from controller seems appropriate given that https://github.com/jenkinsci/jenkins/blob/512f267b4cf504a2641b9a44c0c842322e72c996/core/src/main/java/hudson/slaves/Channels.java#L230 this is apparently run on the same machine as the controller? :scream:

Reply:
It is running on the same machine. It was the same "trick" before (concat and later do split) ... Now this could be avoided, if ClasspathBuilder could directly be added to another ClasspathBuilder instance. 
In general I do not know, why the author of the selenium plugin did not put all the classpath together (like now proposed), but I know this change keeps the plugin running with Java > 8.